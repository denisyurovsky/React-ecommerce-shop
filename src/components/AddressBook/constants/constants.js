export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
