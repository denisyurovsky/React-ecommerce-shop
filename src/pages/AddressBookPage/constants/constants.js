export const MODAL_TYPE = {
  EDIT: 'edit',
  ADD: 'add',
};

export const EMPTY_ADDRESS = {
  title: '',
  name: '',
  surname: '',
  country: { name: '', id: '' },
  city: '',
  street: '',
  building: '',
  flat: '',
  phone: '',
  zip: '',
};

export const LINKS = [
  { url: '/', text: 'Home' },
  { url: '/profile', text: 'Profile' },
  {
    url: `/profile/address-book`,
    text: 'Address book',
  },
];

export const RUSSIA = 'Russian Federation';
