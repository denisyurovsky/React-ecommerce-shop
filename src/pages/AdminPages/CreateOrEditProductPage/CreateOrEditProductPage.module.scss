@import "../../../assets/styles/colors";
@import "../../../assets/styles/breakpoints";

.box {
  background-color: white;
  padding: 30px 20px;
  display: flex;
  flex-direction: column;
  gap: 40px;
  min-height: 100%;
  border-radius: 4px;
  box-shadow: 0 0 4px 0 $greyDark;
}

.breadCrumbs {
  padding-top: 20px;
}

.uploadButton {
  height: 70px;
}

.wrapperForButtons {
  display: flex;
  margin-left: auto;
  gap: 20px;
  height: 100%;
}

@media (min-width: $md) {
  .box {
    padding: 50px 200px;
  }
}
