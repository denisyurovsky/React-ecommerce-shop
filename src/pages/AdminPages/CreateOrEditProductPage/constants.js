export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
