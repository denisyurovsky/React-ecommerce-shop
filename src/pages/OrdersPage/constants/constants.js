export const breadcrumbsLinks = [
  { url: '/', text: 'Home' },
  { url: '/profile', text: 'Profile' },
  { url: '/profile/orders', text: 'My orders' },
];
