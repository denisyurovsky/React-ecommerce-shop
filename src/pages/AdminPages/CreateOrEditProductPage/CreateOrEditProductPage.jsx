import { Container } from '@mui/material';
import { EditorState } from 'draft-js';
import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useLocation, useNavigate, useParams } from 'react-router-dom';
import { toast } from 'react-toastify';

import { createProduct, getProduct, updateProduct } from '../../../api/product';
import BasicBreadcrumbs from '../../../components/Breadcrumbs/Breadcrumbs';
import CreateOrEditProductContent from '../../../components/CreateOrEditProductContent/CreateOrEditProductContent';
import { Title } from '../../../components/Title/Title';
import Spinner from '../../../components/ui-kit/Spinner/Spinner';
import { pathNames } from '../../../constants/pathNames';
import { getCategories } from '../../../store/categories/categoriesSlice';

import styles from './CreateOrEditProductPage.module.scss';

const { ADMIN, PRODUCTS, CREATE } = pathNames;

export const CreateOrEditProductPage = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { id: productId } = useParams();

  const [product, setProduct] = useState({
    name: '',
    category: {
      id: null,
      name: '',
    },
    price: null,
    discountPrice: null,
    description: EditorState.createEmpty(),
    images: [],
  });
  const [isLoading, setIsLoading] = useState(false);
  const isEditPage = useLocation().pathname.includes('edit');

  const links = [
    { url: '/', text: 'Home' },
    {
      url: `${ADMIN}${PRODUCTS}${CREATE}`,
      text: isEditPage ? 'Edit Product' : 'Create New Product',
    },
  ];

  const getProductInfo = async (id) => {
    const response = await getProduct(id);

    setProduct(response);
    setIsLoading(false);
  };

  useEffect(() => {
    dispatch(getCategories());

    if (isEditPage) {
      setIsLoading(true);
      getProductInfo(productId);
    }
  }, [dispatch, isEditPage, productId]);

  const onSubmit = async (newProduct) => {
    try {
      setIsLoading(true);

      if (isEditPage) {
        await updateProduct(productId, newProduct);
      } else {
        await createProduct(newProduct);
      }

      toast.success(
        `Product was successfully ${isEditPage ? 'edited' : 'created'}`
      );
    } catch (e) {
      toast.error('Something went wrong');
    } finally {
      navigate('/admin');
    }
  };

  const onCancel = () => navigate('/admin/products');

  if (isLoading) {
    return <Spinner isAbsolute={true} />;
  }

  return (
    <Container maxWidth="lg">
      <BasicBreadcrumbs className={styles.breadCrumbs} links={links} />
      <Title>{isEditPage ? 'Edit Product' : 'Create New Product'}</Title>
      <CreateOrEditProductContent
        product={product}
        onSubmit={onSubmit}
        onCancel={onCancel}
        isEditPage={isEditPage}
      />
    </Container>
  );
};
