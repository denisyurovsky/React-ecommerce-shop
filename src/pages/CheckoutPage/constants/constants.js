import { pathNames } from '../../../constants/pathNames';

export const LINKS = [
  { url: '/', text: 'Home' },
  {
    url: pathNames.CHECKOUT,
    text: 'Checkout',
  },
];

export const EMPTY_ORDER = {
  userId: null,
  id: null,
  products: [],
  status: null,
  addressId: null,
  deliveryAddress: {},
  deliveryPrice: null,
  totalPrice: null,
  totalDiscountPrice: null,
  totalQuantity: null,
  deliveredAt: null,
  createdAt: null,
  updatedAt: null,
};
