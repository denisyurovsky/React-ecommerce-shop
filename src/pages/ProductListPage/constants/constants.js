import { pathNames } from '../../../constants/pathNames';

export const sortTypes = {
  NEW_FIRST: 'new',
  OLD_FIRST: 'old',
  EXPENSIVE_FIRST: 'expensive',
  CHEAP_FIRST: 'cheap',
};

export const pageView = {
  MODULE_VIEW: 'module',
  LIST_VIEW: 'list',
};

export const sortObj = {
  [sortTypes.NEW_FIRST]: { field: 'createdAt', order: 'desc' },
  [sortTypes.OLD_FIRST]: { field: 'createdAt', order: 'asc' },
  [sortTypes.EXPENSIVE_FIRST]: { field: 'price', order: 'desc' },
  [sortTypes.CHEAP_FIRST]: { field: 'price', order: 'asc' },
};

export const BreadcrumbsLinks = [
  { url: '/', text: 'Home' },
  { url: pathNames.PRODUCTS, text: 'Products' },
];

export const NUMBER_ITEMS_ON_PAGE = 20;

export const MOBILE_WIDTH = 540;

export const defaultParams = {
  entity: 'products',
  filters: null,
  sort: sortObj[sortTypes.NEW_FIRST],
  currentPage: 1,
  itemsPerPage: NUMBER_ITEMS_ON_PAGE,
  q: '',
};
