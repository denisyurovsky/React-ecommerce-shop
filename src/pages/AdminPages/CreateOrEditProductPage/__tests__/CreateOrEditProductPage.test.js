import userEvent from '@testing-library/user-event';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import React from 'react';

import { USER_ROLE } from '../../../../constants/constants';
import { categories } from '../../../../test-utils/dto/categoriesDto';
import { productForPDP } from '../../../../test-utils/dto/productsDto';
import renderWith, {
  screen,
  fireEvent,
  createEvent,
  getByRole,
  waitFor,
} from '../../../../test-utils/renderWith';
import { CreateOrEditProductPage } from '../CreateOrEditProductPage';

const preloadedState = {
  categories: {
    data: categories.map((el) => el.name),
  },
  user: {
    user: {
      id: 1,
      role: USER_ROLE.SELLER,
    },
  },
};

const successfullHandlers = [
  rest.get('/categories', (_, res, ctx) => {
    return res(ctx.json(categories));
  }),
  rest.get('/users', (req, res, ctx) => {
    if (req.url.searchParams.get('id') != 1) {
      return res(ctx.status(400));
    }

    return res(ctx.json([{ firstName: 'Lindsay', lastName: 'Yundt' }]));
  }),
  rest.get('/products/0', (_, res, ctx) => res(ctx.json(productForPDP))),
  rest.patch('/products/0', (_, res, ctx) => res(ctx.status(200))),
  rest.post('/products', (_, res, ctx) => res(ctx.status(201))),
];

const errorHandlers = [
  ...successfullHandlers.slice(0, successfullHandlers.length - 2),
  rest.patch('/products/0', (_, res, ctx) =>
    res(ctx.status(403), ctx.json({}))
  ),
  rest.post('/products', (_, res, ctx) => res(ctx.status(403), ctx.json({}))),
];

const getTextArea = () =>
  getByRole(screen.getByText(/description/i).parentNode, 'textbox');

const pasteText = (text) =>
  createEvent.paste(getTextArea(), {
    clipboardData: {
      types: ['text/plain'],
      getData: () => text,
    },
  });

const fillMandatoryFields = () => {
  userEvent.type(screen.getByLabelText(/name/i), 'name');

  userEvent.click(screen.getByLabelText(/category/i));
  userEvent.click(screen.getByText('Grocery'));

  fireEvent(getTextArea(), pasteText('description'));

  userEvent.type(screen.getByLabelText(/^price/i), '690');
};

describe('Successful response', () => {
  const server = setupServer(...successfullHandlers);

  beforeAll(() => server.listen());
  afterAll(() => server.close());

  describe('after editing', () => {
    beforeEach(() =>
      renderWith(
        <CreateOrEditProductPage />,
        preloadedState,
        'admin/products/:id/edit',
        ['/admin/products/0/edit']
      )
    );

    it('should show loading icon', async () => {
      expect(screen.getByRole('progressbar')).toBeInTheDocument();

      await screen.findByLabelText(/^name$/i);
    });

    it('should show confirmation message after submit', async () => {
      const input = await screen.findByLabelText(/^name$/i);

      userEvent.type(input, 'a{backspace}');
      userEvent.click(screen.getByRole('button', { name: /^edit$/i }));

      await screen.findByRole('alert');

      expect(
        screen.getByText('Product was successfully edited')
      ).toBeInTheDocument();
    });

    it('should navigate to /admin/products after cancel', async () => {
      await screen.findByLabelText(/^name$/i);

      userEvent.click(screen.getByRole('button', { name: /^cancel$/i }));

      await waitFor(() =>
        expect(window.location.pathname).toBe('/admin/products')
      );
    });
  });

  describe('after creation', () => {
    beforeEach(() =>
      renderWith(
        <CreateOrEditProductPage />,
        preloadedState,
        'admin/products/create',
        ['/admin/products/create']
      )
    );

    it('should show create new product title on cretion page', async () => {
      expect.assertions(2);

      await screen.findByLabelText(/^name$/i);

      fillMandatoryFields();
      userEvent.click(screen.getByRole('button', { name: /^create$/i }));

      await screen.findByRole('alert');

      expect(
        screen.getByText('Product was successfully created')
      ).toBeInTheDocument();

      await waitFor(() => expect(window.location.pathname).toBe('/admin'));
    });
  });
});

describe('errors in response', () => {
  const server = setupServer(...errorHandlers);

  beforeAll(() => server.listen());
  afterAll(() => server.close());

  describe('during creation', () => {
    beforeEach(() =>
      renderWith(
        <CreateOrEditProductPage />,
        preloadedState,
        'admin/products/create',
        ['/admin/products/create']
      )
    );

    it('should notify about errors', async () => {
      expect.assertions(2);

      await screen.findByLabelText(/^name$/i);

      fillMandatoryFields();
      userEvent.click(screen.getByRole('button', { name: /^create$/i }));

      await screen.findByRole('alert');

      expect(screen.getByText('Something went wrong')).toBeInTheDocument();
      await waitFor(() => expect(window.location.pathname).toBe('/admin'));
    });
  });

  describe('during editing', () => {
    beforeEach(() =>
      renderWith(
        <CreateOrEditProductPage />,
        preloadedState,
        'admin/products/:id/edit',
        ['/admin/products/0/edit']
      )
    );

    it('should notify about errors', async () => {
      expect.assertions(2);

      const input = await screen.findByLabelText(/^name$/i);

      userEvent.type(input, 'a{backspace}');
      userEvent.click(screen.getByRole('button', { name: /^edit$/i }));

      await screen.findByRole('alert');

      expect(screen.getByText('Something went wrong')).toBeInTheDocument();
      await waitFor(() => expect(window.location.pathname).toBe('/admin'));
    });
  });
});
