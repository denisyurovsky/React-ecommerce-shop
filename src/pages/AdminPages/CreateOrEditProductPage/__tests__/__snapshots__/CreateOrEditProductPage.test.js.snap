// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateOrEditProductPage snapshot should take a snapshot with predefined data 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg container css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="breadcrumbs"
      role="presentation"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/admin/products/create"
            >
              Edit Product
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <h2
      class="header"
      data-testid="title"
    >
      Edit Product
    </h2>
    <form>
      <div
        class="box MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="nameInput"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="mui-3"
            id="mui-3-label"
          >
            Name
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-3"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-173wfxe"
              >
                <span>
                  Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            id="selectLabel"
          >
            Category
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            data-testid="selectInput"
            selected_value=""
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="selectLabel selectLabel"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="selectLabel"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-173wfxe"
              >
                <span>
                  Category
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="wraper MuiBox-root css-0"
          tabindex="-1"
        >
          <fieldset
            class="container"
          >
            <legend
              class="legend"
            >
              Description
            </legend>
            <div
              class="modifiers MuiBox-root css-0"
            >
              <div
                class="css-e53awj-MuiStack-root"
              >
                <div
                  class="MuiBox-root css-0"
                >
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-xawwh6-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="FormatBoldIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.6 10.79c.97-.67 1.65-1.77 1.65-2.79 0-2.26-1.75-4-4-4H7v14h7.04c2.09 0 3.71-1.7 3.71-3.79 0-1.52-.86-2.82-2.15-3.42zM10 6.5h3c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5h-3v-3zm3.5 9H10v-3h3.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-xawwh6-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="FormatItalicIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 4v3h2.21l-3.42 8H6v3h8v-3h-2.21l3.42-8H18V4z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-xawwh6-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="FormatUnderlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17c3.31 0 6-2.69 6-6V3h-2.5v8c0 1.93-1.57 3.5-3.5 3.5S8.5 12.93 8.5 11V3H6v8c0 3.31 2.69 6 6 6zm-7 2v2h14v-2H5z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiBox-root css-12z0wuy"
                >
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-xawwh6-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="FormatListBulletedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 10.5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5zm0-6c-.83 0-1.5.67-1.5 1.5S3.17 7.5 4 7.5 5.5 6.83 5.5 6 4.83 4.5 4 4.5zm0 12c-.83 0-1.5.68-1.5 1.5s.68 1.5 1.5 1.5 1.5-.68 1.5-1.5-.67-1.5-1.5-1.5zM7 19h14v-2H7v2zm0-6h14v-2H7v2zm0-8v2h14V5H7z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-xawwh6-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="FormatListNumberedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M2 17h2v.5H3v1h1v.5H2v1h3v-4H2v1zm1-9h1V4H2v1h1v3zm-1 3h1.8L2 13.1v.9h3v-1H3.2L5 10.9V10H2v1zm5-6v2h14V5H7zm0 14h14v-2H7v2zm0-6h14v-2H7v2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiBox-root css-0"
                >
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-15jdbw8-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="AddLinkIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8 11h8v2H8zm12.1 1H22c0-2.76-2.24-5-5-5h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1zM3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM19 12h-2v3h-3v2h3v3h2v-3h3v-2h-3z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiBox-root css-0"
                >
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-15jdbw8-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="MailIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiBox-root css-0"
                >
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-15jdbw8-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="TableChartIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 10.02h5V21h-5zM17 21h3c1.1 0 2-.9 2-2v-9h-5v11zm3-18H5c-1.1 0-2 .9-2 2v3h19V5c0-1.1-.9-2-2-2zM3 19c0 1.1.9 2 2 2h3V10H3v9z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiStepConnector-root MuiStepConnector-horizontal css-joaw1x-MuiStepConnector-root"
              >
                <span
                  class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
                />
              </div>
            </div>
            <div
              class="editor MuiBox-root css-0"
            >
              <div
                class="DraftEditor-root"
              >
                <div
                  class="DraftEditor-editorContainer"
                >
                  <div
                    class="notranslate public-DraftEditor-content"
                    contenteditable="true"
                    role="textbox"
                    spellcheck="true"
                    style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
                  >
                    <div
                      data-contents="true"
                    >
                      <div
                        class=""
                        data-block="true"
                        data-editor="123"
                        data-offset-key="123-0-0"
                      >
                        <div
                          class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                          data-offset-key="123-0-0"
                        >
                          <span
                            data-offset-key="123-0-0"
                          >
                            <br
                              data-text="true"
                            />
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="footer MuiBox-root css-0"
              data-testid="editor-footer"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 symbols css-ahj2mt-MuiTypography-root"
              >
                Total: 
                <span
                  class=""
                  data-testid="left-number"
                >
                  1000
                </span>
              </p>
            </div>
          </fieldset>
        </div>
        <label
          for="images-upload"
        >
          <input
            accept="image/*"
            class="css-1hyfx7x"
            id="images-upload"
            multiple=""
            type="file"
          />
          <span
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButtonBase-root uploadButton css-xsqe0f-MuiButtonBase-root-MuiButton-root"
            data-testid="imageInput"
            role="button"
            tabindex="0"
          >
            Choose images to upload
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
        </label>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="priceInput"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="mui-4"
            id="mui-4-label"
          >
            Price
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-4"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-173wfxe"
              >
                <span>
                  Price
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="wrapperForButtons MuiBox-root css-0"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="reset"
          >
            Cancel
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Edit
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
